refract-spec.md

```diff
@@ -137,9 +139,481 @@ This is how it would expressed in the compact version.
 
+## Primitive Types
+
+Primitive Types extend upon the Refract Element to create elements for the following types:
+
+- null
+- string
+- number
+- boolean
+- array
+- object
+
+A Refract document MUST support each of these element types by default. These types MAY be extended or replaced based on namespaces provided in any root or parent element.
+
+### Null Element (Element)
+
+A `Null Element` is a type that simply has `null` as its value. It extends upon the `Refract Element` and defines a specific content of `null`.
+
+#### Properties
+- `element`: null (string, fixed)
+- `content`: null (null, fixed)
+
+#### Example
+
+In JSON, `null` is represented as `null`.
+
+```json
+null
+```
+
+In Refract, it is represented as a `Null Element` element.
+
+```json
+{
+  "element": "null",
+  "attributes": {},
+  "content": null
+}
+```
+
+### String Element (Element)
+
+A `String Type` allows for string values.
+
+#### Properties
+- `element`: string (string, fixed)
+- `content` (string, required)
+
+#### Examples
+
+In JSON, an example of a string is:
+
+```json
+"foobar"
+```
+
+In Refract, this string is expanded to a `String Type`.
+
+```json
+{
+  "element": "string",
+  "attributes": {},
+  "content": "foobar"
+}
+```
+
+### Number Element (Element)
+
+A `Number Element` allows for values that are numbers.
+
+#### Properties
+- `element`: number (string, fixed)
+- `content` (number, required)
+
+#### Examples
+
+In JSON, a number is represented as:
+
+```json
+400
+```
+
+In Refract, this number is expanded to a `Number Element`.
+
+```json
+{
+  "element": "number",
+  "attributes": {},
+  "content": 400
+}
+```
+### Boolean Element (Element)
+
+A `Boolean Element` allows for values that are JSON boolean.
+
+#### Properties
+- `element`: boolean (string, fixed)
+- `content` (boolean, required)
+
+#### Examples
+
+In JSON, an example of a boolean is:
+
+```json
+true
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "boolean",
+  "attributes": {},
+  "content": true
+}
+```
+
+### Array Type (Element)
+
+An `Array Type` allows for JSON arrays.
+
+#### Properties
+- `element`: array (string, fixed)
+- `content` (array[Refract Element], required)
+
+
+#### Examples
+
+In JSON, an example of an array is:
+
+```json
+["foo", 400, true]
+```
+
+In Refract, this boolean is expanded to a `Boolean Element`.
+
+```json
+{
+  "element": "array",
+  "attributes": {},
+  "content": [
+    {
+      "element": "string",
+      "attributes": {},
+      "content": "foo"
+    },
+    {
+      "element": "number",
+      "attributes": {},
+      "content": 400
+    },
+    {
+      "element": "boolean",
+      "attributes": {},
+      "content": true
+    }
+  ]
+}
+```
+
+### Property Element (Element)
+
+A `Property Type` allows for values that SHOULD be used in `Object Element` elements.
+
+#### Properties
+- `element`: property (string, fixed)
+- `attributes`
+    - `name` (string, required) - Name of property
```

Comment:
I could make a case for the rule that if there is only an id, that it defines the name. Otherwise I may have to do id=thing name=thing. Would be nice to be able to do id=thing in the case I want to reference and define the property name. Name alone could preserve name with different id or allow anonymity?